crates/rollup-boost/src/flashblocks/inbound.rs

```diff
@@ -35,55 +43,97 @@ enum FlashblocksReceiverError {
 pub struct FlashblocksReceiverService {
     url: Url,
     sender: mpsc::Sender<FlashblocksPayloadV1>,
-    reconnect_ms: u64,
+    websocket_config: FlashblocksWebsocketConfig,
     metrics: FlashblocksWsInboundMetrics,
 }
 
 impl FlashblocksReceiverService {
-    pub fn new(url: Url, sender: mpsc::Sender<FlashblocksPayloadV1>, reconnect_ms: u64) -> Self {
+    pub fn new(
+        url: Url,
+        sender: mpsc::Sender<FlashblocksPayloadV1>,
+        websocket_config: FlashblocksWebsocketConfig,
+    ) -> Self {
         Self {
             url,
             sender,
-            reconnect_ms,
+            websocket_config,
             metrics: Default::default(),
         }
     }
 
     pub async fn run(self) {
+        let mut backoff = self.websocket_config.backoff();
         loop {
-            if let Err(e) = self.connect_and_handle().await {
-                error!("Flashblocks receiver connection error, retrying in 5 seconds: {e}");
+            if let Err(e) = self.connect_and_handle(&mut backoff).await {
+                let interval = backoff
+                    .next_backoff()
+                    .expect("max_elapsed_time not set, never None");
+                error!(
+                    "Flashblocks receiver connection error, retrying in {}ms: {}",
+                    interval.as_millis(),
+                    e
+                );
                 self.metrics.reconnect_attempts.increment(1);
                 self.metrics.connection_status.set(0);
-                tokio::time::sleep(std::time::Duration::from_millis(self.reconnect_ms)).await;
+                tokio::time::sleep(interval).await;
             } else {
                 break;
             }
         }
     }
 
-    async fn connect_and_handle(&self) -> Result<(), FlashblocksReceiverError> {
+    async fn connect_and_handle(
+        &self,
+        backoff: &mut ExponentialBackoff,
+    ) -> Result<(), FlashblocksReceiverError> {
         let (ws_stream, _) = connect_async(self.url.as_str()).await?;
         let (mut write, mut read) = ws_stream.split();
 
+        // if we have successfully connected - reset backoff
+        backoff.reset();
+
         info!("Connected to Flashblocks receiver at {}", self.url);
         self.metrics.connection_status.set(1);
 
         let cancel_token = CancellationToken::new();
         let cancel_for_ping = cancel_token.clone();
 
-        let ping_task = tokio::spawn(async move {
-            let mut ping_interval = interval(Duration::from_millis(500));
+        let ping_set = Arc::new(DashSet::with_capacity(10));
+        let pong_set = ping_set.clone();
+        let cleaning_set = ping_set.clone();
 
+        tokio::spawn(async move {
```

Comment:
instead of an extra task to clean up pings, can just use an LRU cache? with the default settings it would start cleaning up pings ~15 seconds without a pong